### PR TITLE
(Fix) - No eth field for payable functions on tx builder

### LINKED ISSUE #75: Transaction builder: the app is crashed on the "empty" line in the functions list
### Title/Description
The Transaction builder app is crashed with the error "Cannot read property 'map' of undefined" in the console when "empty" ( assuming Fallback function) is selected for the contract `0xc778417e063141139fce010982780140aa0cd5ab`

### Environment
 - Browser: Chrome
 - Safe: any
 - Wallet: MM
 - App Version: 2.14.1
 - Environment:
   - staging (rinkeby)

### Steps to reproduce
1. Go to the Transaction builder
2. Enter contract address `0xc778417e063141139fce010982780140aa0cd5ab`
3. Select "empty" function under deposit
Current result: The app is crashed

### Expected result
The app should not be crashed.

### Screenshots
image

### PR DESCRIPTION
**Pull request by Agupane**
Closes #75 and #74 by:

- Excluding from the abi the methods that aren't of type `function`

Also fixes [#1605](https://github.com/gnosis/safe-react/issues/1607) by:

- Checking that the method is `payable` or not in order to enable the `ETH` input

### CODE REVIEW

File: apps/tx-builder/src/hooks/useServices/interfaceRepository.ts [hunk 1]
```diff
@@ -72,6 +72,10 @@ class InterfaceRepository {
           return false;
         }
 
+        if(e?.type.toLowerCase() !== 'function') {
```
Comment (rmeissner) [inline apps/tx-builder/src/hooks/useServices/interfaceRepository.ts:75]: We should also support `fallback` and `receive` functions (see https://docs.soliditylang.org/en/v0.7.5/abi-spec.html#json)

Comment (rmeissner) [reply apps/tx-builder/src/hooks/useServices/interfaceRepository.ts:75]: This should also impact the eth field

Comment (Agupane) [reply apps/tx-builder/src/hooks/useServices/interfaceRepository.ts]: Thanks for the feedback! I improved it now

File: apps/tx-builder/src/hooks/useServices/interfaceRepository.ts [hunk 2]
```diff
@@ -72,9 +72,16 @@ class InterfaceRepository {
...
 
+        if(e?.type.toLowerCase() === 'event') {
+          return false
+        }
+
         return !e.constant;
       })
       .map((m: any) => {
+        if(m.type === 'fallback' || m.type === 'receive') {
+          return { inputs: [], name: m.type === 'fallback'  ? 'fallback' : 'receive', payable: m.payable || false };
```
Comment (rmeissner) [inline apps/tx-builder/src/hooks/useServices/interfaceRepository.ts:83]: Could we simplify `m.type === 'fallback'  ? 'fallback' : 'receive'` to `m.type`, as we check before that it is either one of those two strings?

**Review by rmeissner — APPROVED** [review]

File: apps/tx-builder/src/hooks/useServices/interfaceRepository.ts (hunk 2 above)
Comment (Agupane) [reply apps/tx-builder/src/hooks/useServices/interfaceRepository.ts]: I improved it, please check it

Comment (rmeissner) [reply apps/tx-builder/src/hooks/useServices/interfaceRepository.ts]: looks good

**Review by dasanra — APPROVED** [review]

**Review by nicosampler — APPROVED** [review]

**Pull request merged**